src/test/java/org/folio/rest/api/HoldingsStorageTest.java

```diff
@@ -1969,6 +1969,77 @@ public void canSearchByDiscoverySuppressProperty() throws Exception {
       containsInAnyOrder(notSuppressedHolding.getId(), notSuppressedHoldingDefault.getId()));
   }
 
+  @Test
+  public void shouldFindHoldingByCallNumberWhenThereIsSuffix() throws Exception {
+    final IndividualResource instance = instancesClient
+      .create(smallAngryPlanet(UUID.randomUUID()));
+
+    final IndividualResource firstHoldingsToMatch = holdingsClient.create(
+      new HoldingRequestBuilder()
+        .forInstance(instance.getId())
+        .withCallNumber("GE77 .F73 2014")
+        .withPermanentLocation(mainLibraryLocationId));
+
+    final IndividualResource secondHoldingsToMatch = holdingsClient.create(
+      new HoldingRequestBuilder()
+        .forInstance(instance.getId())
+        .withCallNumber("GE77 .F73 2014")
+        .withCallNumberSuffix("Curriculum Materials Collection")
+        .withPermanentLocation(mainLibraryLocationId));
+
+    holdingsClient.create(
+      new HoldingRequestBuilder()
+        .forInstance(instance.getId())
+        .withCallNumber("GE77 .F73 ")
+        .withCallNumberSuffix("2014 Curriculum Materials Collection")
+        .withPermanentLocation(mainLibraryLocationId));
+
+    final List<UUID> foundHoldings = searchByCallNumberEyeReadable("GE77 .F73 2014");
+
+    assertThat(foundHoldings.size(), is(2));
+    assertThat(foundHoldings, hasItems(firstHoldingsToMatch.getId(),
+      secondHoldingsToMatch.getId()));
+  }
+
+  @Test
+  public void explicitRightTruncationCanBeApplied() throws Exception {
+    final IndividualResource instance = instancesClient
+      .create(smallAngryPlanet(UUID.randomUUID()));
+
+    final IndividualResource firstHoldingsToMatch = holdingsClient.create(
+      new HoldingRequestBuilder()
+        .forInstance(instance.getId())
+        .withCallNumber("GE77 .F73 2014")
+        .withPermanentLocation(mainLibraryLocationId));
+
+    final IndividualResource secondHoldingsToMatch = holdingsClient.create(
+      new HoldingRequestBuilder()
+        .forInstance(instance.getId())
+        .withCallNumber("GE77 .F73 2014")
+        .withCallNumberSuffix("Curriculum Materials Collection")
+        .withPermanentLocation(mainLibraryLocationId));
+
+    final IndividualResource thirdHoldingsToMatch = holdingsClient.create(
+      new HoldingRequestBuilder()
+        .forInstance(instance.getId())
+        .withCallNumber("GE77 .F73 ")
+        .withCallNumberSuffix("2014 Curriculum Materials Collection")
+        .withPermanentLocation(mainLibraryLocationId));
+
+    holdingsClient.create(
+      new HoldingRequestBuilder()
+        .forInstance(instance.getId())
+        .withCallNumber("GE77 .F74 ")
```

Comment:
Is this intentionally meant to be `.F74` instead of `F73`?

How does the inclusion of this record help describe the scenario better?

Reply:
It demonstrates that filtering really works vs just returns everything from DB. It was added to demonstrate that this record is being excluded from the search result.

Reply:
Ok